### PR TITLE
chore: enum dummy ID for `FuncId` in monomorphizer and docstring fixes

### PR DESCRIPTION
**Pull request by michaeljklein**
# Description

## Problem\*

Small issues or typos encountered during https://github.com/noir-lang/noir/issues/5361

## Documentation\*

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by vezenovm** [comment]
I'm seeing some leftover CI failures. This PR also looks to rename `EnableSideEffects`?

**Comment by michaeljklein** [comment]
@vezenovm yes, exactly. I just patched those failures and it's working locally

File: compiler/noirc_frontend/src/monomorphization/ast.rs [hunk 1]
```diff
@@ -65,7 +67,20 @@ pub struct LocalId(pub u32);
 
 /// A function ID corresponds directly to an index of `Program::functions`
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct FuncId(pub u32);
+// pub struct FuncId(pub u32);
```
Comment (vezenovm) [inline compiler/noirc_frontend/src/monomorphization/ast.rs:70]:
```suggestion
```

File: compiler/noirc_frontend/src/monomorphization/ast.rs [hunk 2]
```diff
@@ -349,18 +364,18 @@ impl Program {
     }
 
     pub fn main_id() -> FuncId {
-        FuncId(0)
+        FuncId::Interned(0)
     }
 
     pub fn take_main_body(&mut self) -> Expression {
-        self.take_function_body(FuncId(0))
+        self.take_function_body(FuncId::Dummy)
```
Comment (vezenovm) [inline compiler/noirc_frontend/src/monomorphization/ast.rs:371]:
```suggestion
        self.take_function_body(self.main_id())
```
Not `0` here? Looks like this method is unused either way

**Review by vezenovm — APPROVED** [review]

File: compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/acir_variable.rs [hunk 3]
```diff
@@ -492,7 +492,7 @@ impl<F: AcirField> AcirContext<F> {
             self.sub_var(sum, mul)
         } else {
             // Implement OR in terms of AND
-            // (NOT a) NAND (NOT b) => a OR b
+            // (NOT a) AND (NOT b) => a OR b
```
Comment (jfecher) [inline compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/acir_variable.rs:495]:
```suggestion
            // (NOT a) AND (NOT b) => NOT (a OR b)
```
`!a && !b` would equal `!(a || b)`, not `a || b`. The original NAND is correct. I'm guessing we just missed a NOT around the result

File: compiler/noirc_frontend/src/monomorphization/ast.rs [hunk 4]
```diff
@@ -349,18 +364,18 @@ impl Program {
...
     }
 
     /// Takes a function body by replacing it with `false` and
     /// returning the previous value
     pub fn take_function_body(&mut self, function: FuncId) -> Expression {
-        let main = &mut self.functions[function.0 as usize];
-        let replacement = Expression::Literal(Literal::Bool(false));
+        let main = &mut self[function];
```
Comment (jfecher) [inline compiler/noirc_frontend/src/monomorphization/ast.rs:377]: We should rename this to `function` while we're changing this code

File: compiler/noirc_frontend/src/monomorphization/ast.rs [hunk 5]
```diff
@@ -65,7 +67,20 @@ pub struct LocalId(pub u32);
 
 /// A function ID corresponds directly to an index of `Program::functions`
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct FuncId(pub u32);
+// pub struct FuncId(pub u32);
+pub enum FuncId {
+    Dummy,
+    Interned(u32),
+}
```
Comment (jfecher) [inline compiler/noirc_frontend/src/monomorphization/ast.rs:74]: It looks like `FuncId::Dummy` is unused after the removal in take_main_body?

**Review by jfecher — CHANGES_REQUESTED** [review]

File: compiler/noirc_frontend/src/monomorphization/ast.rs (hunk 5 above)
Comment (michaeljklein) [reply compiler/noirc_frontend/src/monomorphization/ast.rs]: Removed :+1:

**Review by jfecher — APPROVED** [review]

**Pull request merged**